app/src/main/kotlin/github/com/rhacco/dota2androidapp/viewmodel/MatchesViewModel.kt

```diff
@@ -56,24 +62,19 @@ class MatchesViewModel(application: Application) : AndroidViewModel(application)
                                 newItemData.mTitle = App.instance.getString(
                                         R.string.heading_live_ranked_match, averageMMR, result.match.matchid)
                             if (result.teams?.size == 2) {
-                                val radiantPlayers = result.teams[0].players
-                                val direPlayers = result.teams[1].players
-                                if (radiantPlayers != null && direPlayers != null &&
-                                        radiantPlayers.size == 5 && direPlayers.size == 5) {
-                                    newItemData.mRadiantPlayer0 = radiantPlayers[0].name
-                                    newItemData.mRadiantPlayer1 = radiantPlayers[1].name
-                                    newItemData.mRadiantPlayer2 = radiantPlayers[2].name
-                                    newItemData.mRadiantPlayer3 = radiantPlayers[3].name
-                                    newItemData.mRadiantPlayer4 = radiantPlayers[4].name
-                                    newItemData.mDirePlayer0 = direPlayers[0].name
-                                    newItemData.mDirePlayer1 = direPlayers[1].name
-                                    newItemData.mDirePlayer2 = direPlayers[2].name
-                                    newItemData.mDirePlayer3 = direPlayers[3].name
-                                    newItemData.mDirePlayer4 = direPlayers[4].name
-                                    mLiveMatchesItemDataQuery.value = newItemData
-                                } else
-                                    Log.d(App.instance.getString(R.string.log_msg_debug),
-                                            "GetRealtimeStats returned corrupted players data")
+                                val playerSteamIds = mutableListOf<Long>()
+                                for (team in result.teams) {
+                                    if (team.players?.size == 5)
```

Comment:
Is there ever a time when the # of players is < 5?

Reply:
*Probably* no, but a quick safety check should never be bad for sustainable code. A remote API might malfunction or change over time which we have no control over, these few extra lines can help a lot